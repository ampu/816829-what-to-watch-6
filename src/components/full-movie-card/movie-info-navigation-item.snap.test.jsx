import React from 'react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';

import {render} from '@testing-library/react';

import MovieInfoNavigationItem from './movie-info-navigation-item';

describe(`MovieInfoNavigationItem component`, () => {
  it(`Should render according to the snapshot with no path match`, () => {
    const {container} = render((
      <Router history={createMemoryHistory()}>
        <MovieInfoNavigationItem title="fakeTitle" path="/fakePath"/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot with path match`, () => {
    const history = createMemoryHistory();
    history.push(`/fakePath`);

    const {container} = render((
      <Router history={history}>
        <MovieInfoNavigationItem title="fakeTitle" path="/fakePath"/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });
});
