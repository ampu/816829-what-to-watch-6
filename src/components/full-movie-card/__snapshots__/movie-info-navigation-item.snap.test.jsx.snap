// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfoNavigationItem component Should render according to the snapshot with no path match 1`] = `
<div>
  <li
    class="movie-nav__item"
  >
    <a
      class="movie-nav__link"
      href="/fakePath"
    >
      fakeTitle
    </a>
  </li>
</div>
`;

exports[`MovieInfoNavigationItem component Should render according to the snapshot with path match 1`] = `
<div>
  <li
    class="movie-nav__item movie-nav__item--active"
  >
    <a
      class="movie-nav__link"
      href="/fakePath"
    >
      fakeTitle
    </a>
  </li>
</div>
`;
